packages/web3-eth-abi/src/AbiCoder.js

```diff
@@ -229,7 +229,7 @@ export default class AbiCoder {
             nonIndexedInputItems.push(input);
         });
 
-        if (data) {
+        if (data && data !== '0x') {
```

Comment:
Could you handle the '0x' case here: https://github.com/ethereum/web3.js/blob/1.0/packages/web3-eth-contract/src/decoders/EventLogDecoder.js#L50 

Thanks for the PR!

Reply:
Would something like
`if (response.data === '0x') delete response.data` suffice?
Is that what you meant?

Reply:
I would do it this way:
``` javascript 
if (response.data === '0x') {
reponse.data = null;
}
```

Reply:
Done 👍